transform/transformer.go

```diff
@@ -233,12 +235,24 @@ func (t *LogTransformer) transformDiff() {
 	t.LogContent = strings.Join(transformedLines, "\n")
 }
 
-// truncate to avoid Message:Body is too long (maximum is 65536 characters)
+// truncate to avoid Message:Body is too long (maximum is set per VCS)
 func (t *LogTransformer) truncate() {
+	var maxCommentLength int
+	if t.Vcs == config.VcsGithubEnterprise && t.GithubMaxCommentLength != 0 {
+		maxCommentLength = t.GithubMaxCommentLength
+	} else if t.Vcs == config.VcsGithub || t.Vcs == config.VcsGithubEnterprise {
+		maxCommentLength = provider.GithubMaxCommentLength
+	} else if t.Vcs == config.VcsBitbucket {
+		maxCommentLength = provider.BitbucketMaxCommentLength
+	} else if t.Vcs == config.VcsGitlab {
+		maxCommentLength = provider.GitlabMaxCommentLength
+	}
+	truncatedCommentSuffix := "\n```\n</details>\n<br>\n\n**Warning**: Truncated output as length greater than max comment size."
+	maxCommentLength = maxCommentLength - len([]rune(truncatedCommentSuffix))
```

Comment:
@joshmyers please run go test locally `go test -short ./... -v` I unfortunately can not trigger the circleci tests for forks. 

```
--- FAIL: TestNewLogTransformer (0.00s)
panic: runtime error: slice bounds out of range [:-92] [recovered]
        panic: runtime error: slice bounds out of range [:-92]
```
If the length of truncatedCommentSuffix is greater than maxCommentLength, then maxCommentLength will become negative.

Reply:
Ah, I've been testing locally with:

```
❯ go test -v ./... -run 'TestLogTransform_Truncate'
?   	github.com/karlderkaefer/cdk-notifier	[no test files]
?   	github.com/karlderkaefer/cdk-notifier/cmd	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/karlderkaefer/cdk-notifier/config	(cached) [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/karlderkaefer/cdk-notifier/provider	(cached) [no tests to run]
=== RUN   TestLogTransform_Truncate
--- PASS: TestLogTransform_Truncate (0.07s)
PASS
ok  	github.com/karlderkaefer/cdk-notifier/transform	0.069s
```

but yeh, the other test broke because it isn't setting a provider (github/bitbucket/gitlab etc), fixing.